src/main/java/org/folio/rest/impl/OaiPmhViewInstancesAPI.java

```diff
@@ -36,23 +42,33 @@
     .lookupClass());
 
   private static final String SQL = "select * from pmh_view_function($1,$2,$3,$4);";
+  private static final String SQL_UPDATED_INSTANCES_IDS = "select * from pmh_get_updated_instances_ids($1,$2,$3,$4);";
+  private static final String SQL_INSTANCES = "select * from pmh_instance_view_function($1,$2);";
 
+  @Validate
   @Override
   public void getOaiPmhViewInstances(String startDate, String endDate, boolean deletedRecordSupport,
       boolean skipSuppressedFromDiscoveryRecords, String lang, RoutingContext routingContext, Map<String, String> okapiHeaders,
       Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
 
-    log.debug("request params:", Iterables.toString(routingContext.request()
-      .params()));
+    getCommonInstanceProcessing(SQL,
+      () -> createPostgresParams(startDate, endDate, deletedRecordSupport, skipSuppressedFromDiscoveryRecords),
+      routingContext, okapiHeaders, asyncResultHandler, vertxContext,
+      "Select from oai pmh view completed successfully");
+  }
+
+  private void getCommonInstanceProcessing(String sql, Supplier<Tuple> paramsSupplier, RoutingContext routingContext, Map<String, String> okapiHeaders,
```

Comment:
Maybe rename the method?
```suggestion
  private void fetchInstancesByQuery(String sql, Supplier<Tuple> paramsSupplier, RoutingContext routingContext, Map<String, String> okapiHeaders,
```

Reply:
It looks like you mean that even internally used method needs to be more self-explanatoryform semantic point of view. In this case taking into account thta this "common processing" part of code intended to return no only instance relatd data but also items and holdings and all these entities related to "oai records" conceptually. So possibly the name like "fetchRecordsByQuery" would be more appropriate in this case. I intentionally skipped part of name like "OaiPmh" ("getOaiPmh....") because the code itself belong to module contained this in name.

Reply:
Yes, `fetchRecordsByQuery` works for me as well.

Reply:
Thank you.